File: Algorithm/Programmers/Level1/solution(25).py
# 소수 찾기

def solution(n):
    answer = 0
    if n >= 2 :
        answer += 1
        for i in range(3, n+1, 2): # 짝수 제외
            check = True
            if (int(i**0.5))**2 == n: # 제곱근 제외
                check = False
            else :
                for j in range(3, int(i**0.5+1), 2): # 제곱근 근접 숫자까지만 확인, 짝수 제외
                    if i%j == 0:
                        check = False
                        break
            if check:
                answer += 1
    return answer

File: Algorithm/Programmers/Level1/solution(25).java
// 소수 찾기
// 에라토스테네스 체 활용

class Solution {
    public int solution(int n) {
        int answer = 0;
        int[] check = new int[n+1]; // 0으로 초기화
        
        for (int i = 2; i < n+1; i++){
            if (check[i] == 0){
                answer += 1;
                for (int j = i; j < n+1;){
                    check[j] = 1;
                    j += i;
                }
            }
        }
        
        return answer;
    }
}